nemo/collections/asr/parts/submodules/rnnt_greedy_decoding.py

```diff
@@ -679,8 +671,21 @@ def _greedy_decode_blank_as_pad(
                     if time_idx == 0 and symbols_added == 0 and hidden is None:
                         g, hidden_prime = self._pred_step(self._SOS, hidden, batch_size=batchsize)
                     else:
-                        # Perform batch step prediction of decoder, getting new states and scores ("g")
-                        g, hidden_prime = self._pred_step(last_label, hidden, batch_size=batchsize)
+                        if self.decoder.blank_as_pad:
```

Comment:
Please revert, we dont want to merge these two functions.

Reply:
Can you please explain why we don't want to merge them? They are different in just a few lines, which can be handled with an if-else block (or a new function if you'd prefer that way). And I had to apply the same fix to both functions, so I thought it makes sense to unify them. If we unify them and in the future we need to make some changes, it's going to be in a single place and not in two places, which is easy to forget.

Reply:
They share similar logic but not concept. Normally the prednet is not supposed to have the blank token in it's vocab. We abuse the notation by using the pytorch embeddings mechanism for masking one token in NeMo for efficiency sake but the academically correct code is the second one. 

They should not be merged.

Reply:
Rolled-back